client/src/components/Gallery/Gallery.js

```diff
@@ -27,13 +26,17 @@ export default class Gallery extends React.Component {
     }
 
     render() {
-        const images = JSON.parse(JSON.stringify(this.state.images));
+        const images = JSON.parse(JSON.stringify(this.props.images));
+        let primaryImage = images.find(e => e.isPrimary === true);
+
+        primaryImage = primaryImage !== undefined ?
+            primaryImage :
+            images.splice(0, 1);
```

Comment:
You can just do
```javascript
primaryImage = primaryImage || images.splice(0, 1)[0];
```
The reason being that JS will assign primaryImage to primaryImage if it's truthy (not undefined, null, 0, or false), or if it's falsy it will continue to the other side of the or and assign whatever the return value of that is

Also `splice` returns an array even if it's one element, which is why I had `[ primaryImage ] = images.splice(0, 1);` which is array destructuring and assigns the first index of the array to primaryImage

Reply:
Thank you for your wisdom kind wizard.